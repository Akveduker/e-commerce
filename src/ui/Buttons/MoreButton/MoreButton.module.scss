@import "../../../styles/_vars.scss";
.button {
    border-radius: $borderRadius;
    background-color: $C1F;
    padding: 12.5px 16px;
    display: flex;
    align-items: center;
    cursor: pointer;
    &--active {
        .icon {
            transform: rotate(180deg);
        }
    }
}
.icon {
    width: 16px;
    height: 16px;
    margin-left: 6px;
}
