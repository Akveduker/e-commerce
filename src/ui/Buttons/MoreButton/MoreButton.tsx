import React, { FC } from 'react';
import ArrowDown from '../../icons/ArrowDown/ArrowDown';
import s from './MoreButton.module.scss'
interface MoreButtonProps {
    setFunction: () => void;
    isOpen: boolean;
}
const MoreButton: FC<MoreButtonProps> = ({ setFunction, isOpen }) => {
    const text = isOpen ? `Скрыть` : `Больше категорий`
    return (
        <button
            className={isOpen ? `${s['button']} ${s['button--active']}` : `${s['button']}`}
            onClick={setFunction}
            type='button'
            data-testid={'toggle-button'}
        >
            {text}
            <ArrowDown className={s['icon']} />
        </button>
    );
};

export default MoreButton;