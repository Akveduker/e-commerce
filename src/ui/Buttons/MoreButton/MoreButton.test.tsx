import { render, screen } from "@testing-library/react"
import MoreButton from "./MoreButton"

describe('MoreButton', () => {
    describe('Кнопка активна', () => {
        const element = <MoreButton isOpen setFunction={jest.fn()} />
        test('Снапшот активной кнопки', () => {
            const { container } = render(element)
            expect(container).toMatchSnapshot()
        })
        test('Активный текст', () => {
            render(element)
            const button = screen.getByTestId('toggle-button')
            expect(button).toHaveTextContent(/скрыть/i)
        })
        test('Активный класс', () => {
            render(element)
            const button = screen.getByTestId('toggle-button')
            expect(button).toHaveClass('button--active')
        })
    })
    describe('Кнопка неактивна', () => {
        const element = <MoreButton isOpen={false} setFunction={jest.fn()} />
        test('Снапшот неактивной кнопки', () => {
            const { container } = render(element)
            expect(container).toMatchSnapshot()
        })
        test('Неактивный текст', () => {
            render(element)
            const button = screen.getByTestId('toggle-button')
            expect(button).toHaveTextContent(/больше категорий/i)
        })
        test('Неактивный класс', () => {
            render(element)
            const button = screen.getByTestId('toggle-button')
            expect(button).not.toHaveClass('button--active')
        })
    })
})