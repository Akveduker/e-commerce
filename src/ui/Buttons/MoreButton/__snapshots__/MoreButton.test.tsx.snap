// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoreButton Кнопка активна Снапшот активной кнопки 1`] = `
<div>
  <button
    class="button button--active"
    data-testid="toggle-button"
    type="button"
  >
    Скрыть
    <svg
      class="icon"
      fill="none"
      height="16"
      viewBox="0 0 16 16"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M4.81348 6.53336L7.55348 9.27336C7.67838 9.39752 7.84735 9.46722 8.02348 9.46722C8.1996 9.46722 8.36857 9.39752 8.49348 9.27336L11.1601 6.60669"
        stroke="#151515"
        stroke-linecap="round"
        stroke-linejoin="bevel"
        stroke-width="2.5"
      />
    </svg>
  </button>
</div>
`;

exports[`MoreButton Кнопка неактивна Снапшот неактивной кнопки 1`] = `
<div>
  <button
    class="button"
    data-testid="toggle-button"
    type="button"
  >
    Больше категорий
    <svg
      class="icon"
      fill="none"
      height="16"
      viewBox="0 0 16 16"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M4.81348 6.53336L7.55348 9.27336C7.67838 9.39752 7.84735 9.46722 8.02348 9.46722C8.1996 9.46722 8.36857 9.39752 8.49348 9.27336L11.1601 6.60669"
        stroke="#151515"
        stroke-linecap="round"
        stroke-linejoin="bevel"
        stroke-width="2.5"
      />
    </svg>
  </button>
</div>
`;
